apache/zeppelin
# ZEPPELIN-598 ] Dynamic loading for Interpreter and API

### What is this PR for?

Use of external libraries that are included in the Roadmap,
The interpreter needs related to dynamic loading capabilities.
### What type of PR is it?

Feature

### What is the Jira issue?
https://issues.apache.org/jira/browse/ZEPPELIN-598

### How should this be tested?
1. Run to `mvn clean package -DskipTests`
2. Run to `mvn clean package -DskipTests -pl zeppelin-server --am`
3. run zeppelin.
4. move to interpreter menu on web ui.
   - check empty interpreter
5. call restapi
   
```
POST http://localhost:8080/api/interpreter/load/md/md
{"artifact":"org.apache.zeppelin:zeppelin-markdown:0.6.2", "className":"org.apache.zeppelin.markdown.Markdown"}
```

6. check interpreter menu to create interpreter `md` on web ui.
7. add interpreter setting `md`
8. run paragraph `md`
```
%md
Hello markdown
```
9. call unload restapi
```
DELETE http://localhost:8080/api/interpreter/unload/md2/md2/org.apache.zeppelin.markdown.Markdown
```
10. check delete `md` on web ui
11. restart to paragraph for `md` 
     if result is error status then well working.

### Screenshots (if appropriate)
[Click to link for screenshot](https://cloud.githubusercontent.com/assets/10525473/22174189/1b2cce6c-df8d-11e6-975f-333a502079ea.gif)


### Questions:
* Does the licenses files need update? no
* Is there breaking changes for older versions? no
* Does this needs documentation? we have had already.
